vendors/nuvoton/sdk/middleware/wifi_esp8266/esp8266_wifi.c

```diff
@@ -635,19 +776,42 @@ ESP_WIFI_Status_t ESP_WIFI_GetNetStatus( ESP_WIFI_Object_t * pxObj )
 }
 
 /**
- * @brief  Reset the Ipd buffer
+ * @brief  Clear the Ipd buffer
  */
-void ESP_WIFI_Reset_Ipd( ESP_WIFI_Conn_t * pxConn )
+void ESP_WIFI_Clear_Ipd( ESP_WIFI_Object_t * pxObj, uint8_t LinkID )
 {
-    xWifiIpd[pxConn->LinkID].DataLength = 0;
+    struct ESP_WIFI_PACKET **p = &gPackets;
+
+    while (*p) {
+        if ((*p)->LinkID == LinkID || LinkID == ESP8266_ALL_SOCKET_IDS) {
+            struct ESP_WIFI_PACKET *q = *p;
+
+            if (gPacketsEnd == &(*p)->next) {
+                gPacketsEnd = p; // Set last packet next field/gPacketsEnd
+            }
+            *p = (*p)->next;
+            pxObj->HeapUsage -= (sizeof(struct ESP_WIFI_PACKET) + q->TotalLength);;
+            free(q);
```

Comment:
Hi @cyliangtw,

I've just noticed that you are using the standard `free()` function here to free memory created by `pvPortMalloc()`, instead of using `vPortFree()` from the heap_4.c implementation, likely leading to some fragmentation. With this change, all tests now pass:

```suggestion
            vPortFree(q);
```


Could you please make and verify this change? I believe it's the only remaining step for this PR. Thanks!

Reply:
Hi @sukhmanm ,
Thanks of your detection, it's fixed in commit 977a919.